frontend/controller/actions/group.js

```diff
@@ -142,6 +203,146 @@ export default (sbp('sbp/selectors/register', {
+        ...omit(params, ['options']),
+        contractID: params.data.chatRoomID,
+        data: { username },
+        hooks: {
+          prepublish: params.hooks?.prepublish,
+          postpublish: null
+        }
+      })
+
+      if (username === rootState.loggedIn.username) {
+        // 'READY_TO_JOIN_CHATROOM' is necessary to identify the joining chatroom action is NEW or OLD
+        // Users join the chatroom thru group making group actions
+        // But when user joins the group, he needs to ignore all the actions about chatroom
+        // Because the user is joining group, not joining chatroom
+        // and he is going to make a new action to join 'General' chatroom AGAIN
+        // While joining group, we don't set this flag because Joining chatroom actions are all OLD ones, which needs to be ignored
+        // Joining 'General' chatroom is one of the step to join group
+        // So setting 'READY_TO_JOIN_CHATROOM' can not be out of the 'JOINING_GROUP' scope
+        sbp('okTurtles.data/set', 'READY_TO_JOIN_CHATROOM', true)
+      }
+      await sbp('chelonia/out/actionEncrypted', {
+        ...omit(params, ['options']),
+        action: 'gi.contracts/group/joinChatRoom',
+        hooks: {
+          prepublish: null,
+          postpublish: params.hooks?.postpublish
+        }
+      })
+    } catch (e) {
+      console.error('gi.actions/group/joinChatRoom failed!', e)
+      throw new GIErrorUIRuntimeError(L('Failed to join chat channel.'))
+    }
+    return true
+  },
+  'gi.actions/group/addAndJoinChatRoom': async function (params: GIActionParams) {
+    const message = await sbp('gi.actions/group/addChatRoom', {
+      ...omit(params, ['options']),
+      hooks: {
+        prepublish: params.hooks?.prepublish,
+        postpublish: null
+      }
+    })
+
+    await sbp('gi.actions/group/joinChatRoom', {
+      ...omit(params, ['options']),
+      data: {
+        chatRoomID: message.contractID()
+      },
+      hooks: {
+        prepublish: null,
+        postpublish: params.hooks?.postpublish
+      }
+    })
+
+    return message
+  },
+  'gi.actions/group/renameChatRoom': async function (params: GIActionParams) {
+    try {
+      await sbp('gi.actions/chatroom/rename', {
+        ...omit(params, ['options']),
+        contractID: params.data.chatRoomID,
+        data: {
+          name: params.data.name
+        },
+        hooks: {
+          prepublish: params.hooks?.prepublish,
+          postpublish: null
+        }
+      })
+
+      await sbp('chelonia/out/actionEncrypted', {
+        ...omit(params, ['options']),
+        action: 'gi.contracts/group/renameChatRoom',
+        hooks: {
+          prepublish: null,
+          postpublish: params.hooks?.postpublish
+        }
+      })
+    } catch (e) {
+      console.error('gi.actions/group/renameChatRoom failed!', e)
+      throw new GIErrorUIRuntimeError(L('Failed to rename chat channel.'))
+    }
+    return true
```

Comment:
all actions should return the GIMessage for the group contract message that was sent.

So here since we send out two GIMessages, return the second one for `'gi.contracts/group/renameChatRoom'` since it corresponds to the `gi.actions/group` action.